hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/AsyncFSWAL.java

```diff
@@ -323,7 +324,9 @@ private void syncFailed(long epochWhenSync, Throwable error) {
   private void syncCompleted(AsyncWriter writer, long processedTxid, long startTimeNs) {
     highestSyncedTxid.set(processedTxid);
     for (Iterator<FSWALEntry> iter = unackedAppends.iterator(); iter.hasNext();) {
-      if (iter.next().getTxid() <= processedTxid) {
+      FSWALEntry entry = iter.next();
+      if (entry.getTxid() <= processedTxid) {
+        entry.getRpcCall().ifPresent(ServerCall::releaseByWAL);
```

Comment:
Still worry about the CPU-consuming in  ServerCall#release, for AsyncFSWAL,  we will accumulate the batch size entries and then call the sync,  so if the sync not finished, then the ServerCall#release will be always looping..

Reply:
Not problem  as said before.